src/nft/components/collection/CollectionNfts.tsx

```diff
@@ -287,6 +288,72 @@ export const CollectionNfts = ({ contractAddress, collectionStats, rarityVerifie
 
   const { assets: collectionNfts, loadNext, hasNext, isLoadingNext } = useLazyLoadAssetsQuery(assetQueryParams)
 
+  const isSelected = useCallback(
+    (asset: GenieAsset) => {
+      return itemsInBag.some((item) => asset.tokenId === item.asset.tokenId && asset.address === item.asset.address)
+    },
+    [itemsInBag]
+  )
+
+  const calculatePrice = useCallback(
+    (asset: GenieAsset) => {
+      return isSelected(asset)
+        ? calcPoolPrice(
+            asset,
+            itemsInBag
+              .filter((item) => item.asset.address === asset.address && item.asset.marketplace === asset.marketplace)
+              .map((item) => item.asset.tokenId)
+              .indexOf(asset.tokenId)
+          )
+        : calcPoolPrice(
+            asset,
+            itemsInBag.filter(
+              (item) => item.asset.address === asset.address && item.asset.marketplace === asset.marketplace
+            ).length
+          )
+    },
+    [isSelected, itemsInBag]
+  )
+
+  const collectionAssets = useMemo(() => {
+    if (
+      !collectionNfts ||
+      !collectionNfts.some((asset) => asset.marketplace === Markets.NFTX || asset.marketplace === Markets.NFT20)
+    ) {
+      return collectionNfts
+    }
+
+    const collectionAssets = [...collectionNfts]
```

Comment:
is this line necessary?

Reply:
yes it does seem to make a difference

Reply:
it should overwrite the line 315 value of `collectionAssets` mutate it conditionally in the `forEach`, then potentially mutate it again with .sort, then return a new value for it that `useMemo` will use. I expect there's some weird variable shadowing issue here. I think you should make this work as you need w/o overwriting collectionAssets. I'm pretty concerned about unexpected buggy behavior with this as written.